packages/react/src/__tests__/FormControl.test.tsx

```diff
@@ -390,6 +391,41 @@ describe('FormControl', () => {
         expect(consoleSpy).toHaveBeenCalled()
         consoleSpy.mockRestore()
       })
+
+      it('should not add required prop to individual checkbox', async () => {
+        const {getByRole} = render(
+          <FormControl required>
```

Comment:
As a follow up comment after my previous message, **if an individual checkbox needs to be in a group to be required**, I wonder if we can control this with Typescript or maybe an eslint rule so that consumers don't get confused when the required prop doesn't show on the DOM tree. 

What do you think?

Reply:
Good observation. Looks like [we're going to allow individually required checkboxes](https://github.com/primer/react/pull/5027#discussion_r1787922994), so the eslint rule won't be needed in this case. I will update the code/tests!